portal/src/graphql/portal/OnboardingConfigAppScreen.tsx

```diff
@@ -0,0 +1,300 @@
+      if (btnItem.loginIDType) {
+        if (checked) {
+          loginIDKeys.add(btnItem.loginIDType);
+        } else {
+          loginIDKeys.delete(btnItem.loginIDType);
+        }
+      } else if (btnItem.identityType) {
+        if (checked) {
+          identities.add(btnItem.identityType);
+        } else {
+          identities.delete(btnItem.identityType);
+        }
+      } else {
+        console.error(
+          "IdentitiesButton should have either identityType or loginIDType"
+        );
+        return;
+      }
+
+      // check if there is any login id enabled
+      // and update login_id in identities list
+      if (loginIDKeys.size > 0) {
+        identities.add("login_id");
+      } else {
+        identities.delete("login_id");
+      }
+
+      setState((prev) => ({
+        ...prev,
+        pendingForm: {
+          ...prev.pendingForm,
+          identities,
+          loginIDKeys,
+        },
+      }));
+    },
+    [state, setState, btnItem]
+  );
+
+  const onItemClick = useCallback(
+    (event: React.FormEvent) => {
+      event.preventDefault();
+      event.stopPropagation();
+      const currentChecked = getCheckedState();
+      onCheckedChange(!currentChecked);
+    },
+    [getCheckedState, onCheckedChange]
+  );
+
+  const onCheckboxChange = useCallback(
+    (event?: React.FormEvent, checked?: boolean) => {
+      event?.preventDefault();
+      event?.stopPropagation();
+      onCheckedChange(checked);
+    },
+    [onCheckedChange]
+  );
+
+  return (
+    <div className={styles.identityListItem} onClick={onItemClick}>
+      <div className={styles.label}>
+        <FontIcon iconName={btnItem.iconName} className={styles.icon} />
+        <span className={styles.text}>
+          <FormattedMessage id={btnItem.labelId} />
+        </span>
+      </div>
+      <Checkbox
+        className={styles.checkbox}
+        checked={getCheckedState()}
+        onChange={onCheckboxChange}
+      />
+    </div>
+  );
+};
+
+interface IdentitiesListContentProps {
+  form: AppConfigFormModel<FormState>;
+}
+
+const IdentitiesListContent: React.FC<IdentitiesListContentProps> = function IdentitiesListContent(
+  props
+) {
+  const { form } = props;
+
+  const identitiesButtonItems: IdentitiesButton[] = useMemo(
+    () => [
+      {
+        labelId: "Onboarding.identities.email",
+        iconName: "Mail",
+        loginIDType: "email",
+      },
+      {
+        labelId: "Onboarding.identities.phone",
+        iconName: "CellPhone",
+        loginIDType: "phone",
+      },
+      {
+        labelId: "Onboarding.identities.username",
+        iconName: "ContactCard",
+        loginIDType: "username",
+      },
+      {
+        labelId: "Onboarding.identities.sso",
+        iconName: "Globe",
+        identityType: "oauth",
+      },
+      {
+        labelId: "Onboarding.identities.anonymous-user",
+        iconName: "FollowUser",
+        identityType: "anonymous",
+      },
+    ],
+    []
+  );
```

Comment:
This array is static. So it can be defined as a module level constant, saving us from using `useMemo`.